compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1469,6 +1470,23 @@ mod signal_handler {
             libc::sigaction(libc::SIGSEGV, &sa, std::ptr::null_mut());
         }
     }
+
+    /// Modern kernels on modern hardware can have dynamic signal stack sizes.
+    #[cfg(any(target_os = "linux", target_os = "android"))]
+    fn min_sigstack_size() -> usize {
+        const AT_MINSIGSTKSZ: core::ffi::c_ulong = 51;
+        let dynamic_sigstksz = unsafe { libc::getauxval(AT_MINSIGSTKSZ) };
```

Comment:
doesn't need to block this PR, but does it make sense to upstream AT_MINSIGSTKSZ to the libc crate so we don't need to hardcode it here?

Reply:
you mean would it be good for https://github.com/rust-lang/libc/pull/3125 to be reviewed and accepted? yes, I think so.